fog/types/src/view.rs

```diff
@@ -51,6 +51,7 @@ pub struct QueryRequest {
 pub struct QueryResponse {
     /// The number of blocks processed by this view enclave at the time that the
     /// request was evaluated.
+    ///  TODO: Rename to highest_observed_block_count
```

Comment:
I think it would be better to keep this named as highest processed block count at the API boundary, because the client needs to know basically, what blocks are we telling them we have fully processed, or alternatively, what is the index of the first block we haven't fully processed

For the internal block tracker object it may be a good idea to rename but in the proto offered to the client, they have a different field `last_known_block_count` which is the size of the blockchain right now, and it may become confusing how that is different from `highest_observed_block_count`. The previous name was good because we're telling the client what we have actually processed